cl/citations/api_views.py

```diff
@@ -0,0 +1,209 @@
+from cl.search.models import OpinionCluster
+from cl.search.selectors import get_clusters_from_citation_str
+
+SLUGIFIED_EDITIONS: dict[str, str] = {
+    str(slugify(item)): item for item in EDITIONS.keys()
+}
+
+
+class CitationLookupView(GenericAPIView):
+    queryset = OpinionCluster.objects.all()
+    pagination_class = MediumAdjustablePagination
+    serializer = OpinionClusterSerializer
+
+    def get(self, request: Request, *args, **kwargs) -> HttpResponse:
+        query = request.query_params
+
+        # Uses the serializer to perform object level validations
+        citation_serializer = CitationRequestSerializer(data=query)
+        citation_serializer.is_valid(raise_exception=True)
+
+        # Get query parameters from the validated data
+        data = citation_serializer.validated_data
+        self.full_text_citation = data.get("text_citation", None)
+        self.reporter = data.get("reporter", None)
+        self.volume = data.get("volume", None)
+        self.page = data.get("citation_page", None)
+
+        if self.full_text_citation:
+            citations = eyecite.get_citations(self.full_text_citation)
+            if not citations:
+                raise NotFound(
+                    f"Unable to find a citation withing the provided text"
+                )
+
+            if citations[0].groups:
+                c = citations[0]
+                self.reporter = slugify(c.groups["reporter"])
+                self.volume = c.groups["volume"]
+                self.page = c.groups["page"]
+            else:
+                raise NotFound(
+                    f"The provided text is not a valid citation. Please review it and try again"
+                )
+
+        self.reporter_slug = slugify(self.reporter)
+
+        # Look up the reporter to get its proper version (so-2d -> So. 2d)
+        proper_reporter = SLUGIFIED_EDITIONS.get(self.reporter_slug, None)
+        if not proper_reporter:
+            proper_reporter = self._attempt_reporter_variation()
+
+        if proper_reporter and self.volume and self.page:
+            return self._citation_handler(request, proper_reporter)
+        else:
+            return self._reporter_volume_handler(request, proper_reporter)
+
+    def _attempt_reporter_variation(self) -> str:
+        """Try to disambiguate an unknown reporter using the variations dict.
+
+        Raises:
+            NotFound: If the unknown reporter string doesn't match a canonical.
+            MultipleChoices: If the reporter variation maps to more than one
+                            canonical reporter.
+
+        Returns:
+            str: canonical name for the reporter.
+        """
+        potential_canonicals = get_canonicals_from_reporter(self.reporter_slug)
+
+        if len(potential_canonicals) == 0:
+            # Couldn't find it as a variation. Give up.
+            raise NotFound(
+                f"Unable to find Reporter with abbreviation of '{self.reporter}'"
```

Comment:
```suggestion
                f"Unable to find reporter with abbreviation of '{self.reporter}'"
```

Reply:
Also, will this value be slugified? Do we need to store the raw value so we can return it properly in this error message?

Reply:
> will this value be slugified? 

No. The error message will display the reporter name provided by the user.. 

> Do we need to store the raw value so we can return it properly in this error message?

Yes, I think it's a good idea to store the raw value. Most citation helpers use a slugified version of the reporter name for internal purposes, but this could be confusing in error messages. That's why we have separate variables for the reporter name and its slug.